src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -596,9 +595,19 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
       this._overlayRef.updateSize({width: this._getPanelWidth()});
     }
 
-    if (this._overlayRef && !this._overlayRef.hasAttached()) {
-      this._overlayRef.attach(this._portal);
+    if (this._currentPortalAttached !== this.autocomplete._portal) {
```

Comment:
I'm not sure why you'd need to do this. You can just call `detach` and then `attach` again to set the new portal.

Reply:
I see your point there, but i am currently working on an issue (appears with and without this check) where a new panel will be created, even if the template has not changed. This especially happens in the tests, thats why a lot of them are failing.

For example take a look at this test: [autocomplete.spec.ts#L267](https://github.com/angular/material2/blob/master/src/lib/autocomplete/autocomplete.spec.ts#L267). 
It is failing because we are creating a new panel on `focusin` and later on `typeInElement` (`_attachOverlay` gets called in multiple times).
So i need to find a way to check if we should detach/reatach (when a new `mat-autocomplete` has been set or not.
Maybe you have a good solution/idea for this one? Detach only if the template has changed?

Reply:
You don't have to do it in `_attachPortal` though. Keeping `_attachPortal` as it is in master, aside from taking the portal from the panel, and calling `detach` when the `autocomplete` input changes should do it.

Reply:
Changed it to a detach in the autocomplete input. Thanks for your help